pkg/app/piped/platformprovider/ecs/client.go

```diff
@@ -216,6 +219,34 @@ func (c *client) CreateTaskSet(ctx context.Context, service types.Service, taskD
 	if err != nil {
 		return nil, fmt.Errorf("failed to create ECS task set %s: %w", *taskDefinition.TaskDefinitionArn, err)
 	}
+
+	// Wait created TaskSet to be stable.
+	waitInput := &ecs.DescribeTaskSetsInput{
+		Cluster:  service.ClusterArn,
+		Service:  service.ServiceArn,
+		TaskSets: []string{*output.TaskSet.TaskSetArn},
+	}
+
+	retry := backoff.NewRetry(retryTaskSetStable, backoff.NewConstant(retryTaskSetStableInterval))
+	_, err = retry.Do(ctx, func() (interface{}, error) {
+		output, err := c.ecsClient.DescribeTaskSets(ctx, waitInput)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get ECS task set %s: %w", *taskDefinition.TaskDefinitionArn, err)
+		}
+		if len(output.TaskSets) == 0 {
+			return nil, fmt.Errorf("failed to get ECS task set %s: task sets empty", *taskDefinition.TaskDefinitionArn)
+		}
+		taskSet := output.TaskSets[0]
+		if taskSet.StabilityStatus == types.StabilityStatusSteadyState {
+			return nil, nil
+		}
```

Comment:
How about having a log entry here to show the current state (e.g. number of tasks and their status: pending or ...)